src/main/msp/msp.c

```diff
@@ -2839,14 +2848,25 @@ static mspResult_e mspProcessInCommand(mspDescriptor_t srcDesc, int16_t cmdMSP,
         motorConfigMutable()->mincommand = sbufReadU16(src);
 
         // version 1.42
-        if (sbufBytesRemaining(src) >= 2) {
+        if (sbufBytesRemaining(src) > 1) {
             motorConfigMutable()->motorPoleCount = sbufReadU8(src);
+
+            // version 1.44
 #if defined(USE_DSHOT_TELEMETRY)
             motorConfigMutable()->dev.useDshotTelemetry = sbufReadU8(src);
 #else
             sbufReadU8(src);
 #endif
         }
+
+        // Version 1.46
+        if (sbufBytesRemaining(src) > 0) {
```

Comment:
This is more explicit (gives a hint about how many bytes you actually need)
```suggestion
        if (sbufBytesRemaining(src) >= 1) {
```

Reply:
fixed